src/collections/componentSet.ts

```diff
@@ -750,3 +740,24 @@ const splitOnFirstDelimiter = (input: string): [string, string] => {
   const indexOfSplitChar = input.indexOf(KEY_DELIMITER);
   return [input.substring(0, indexOfSplitChar), input.substring(indexOfSplitChar + 1)];
 };
+
+const parseFromSourceInputs = (given: string | string[] | FromSourceOptions): FromSourceOptions => {
+  if (Array.isArray(given)) {
+    return { fsPaths: given };
+  } else if (typeof given === 'object') {
+    return given.include ? { ...given, include: removeMarkedForDelete(given.include) } : given;
+  } else {
+    return { fsPaths: [given] };
+  }
+};
+
+/** remove components that are marked for delete */
+const removeMarkedForDelete = (include: ComponentSet): ComponentSet => {
+  const filtered = new ComponentSet([], include.registry);
+  for (const component of include) {
```

Comment:
@mshanemc you could also use `include.getSourceComponents()` instead of `component instanceof SourceComponent`